portal-bridge/src/types/state/config.rs

```diff
@@ -0,0 +1,15 @@
+#[derive(Debug, Clone)]
+pub struct StateConfig {
+    /// This flag when enabled will storage all the trie keys from block execution in a cache, this
+    /// is needed for gossiping the storage trie's changes.
+    pub cache_contract_storage_changes: bool,
```

Comment:
We only want this flag enabled if we are gossiping storage changes. I want to be an execution client so by default this feature for our bridge wouldn't be enabled.

Reply:
> I'm not sure what you mean by "be an execution client", considering that all this code is inside portal-bridge. Can you explain this a bit more?

This code will be moved into a crate called `trin-execution`. If we are able to execute to the head of the head of the chain we built an execution client. If we are building an execution client (which we are as a byproduct of gossiping state at the head of the chain), then we can release our execution client to the public for others to use. It will be an execution client like Besu, Geth, Reth, Nethermind. The difference is we will be building it without devp2p and will never add devp2p. Devp2p is likely to be dropped sooner or later.

Reply:
> Also, I think it would make sense to call it something like gossip_state_changes

This class is only needed for gossip storage trie changes, as the account trie we can gossip it for free. The storage trie's require storing extra stuff in the database, the majority of users won't need this so it is off by default

Reply:
> This code will be moved into a crate called `trin-execution`

Yes, this makes sense. It's something I wanted to suggest as well. Thank you for clarifying.